src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover.js

```diff
@@ -940,13 +1009,24 @@ function discoverController(
     $scope.minimumVisibleRows = $scope.hits;
   };
 
-  $scope.updateSavedQueryId = newSavedQueryId => {
-    if (newSavedQueryId) {
-      $state.savedQuery = newSavedQueryId;
-    } else {
-      delete $state.savedQuery;
-    }
-    $state.save();
+  $scope.onSavedQuerySaved = savedQuery => {
+    $scope.savedQuery = savedQuery;
+    updateStateFromSavedQuery(savedQuery);
+  };
+
+  $scope.onSavedQueryUpdated = savedQuery => {
+    $scope.savedQuery = { ...savedQuery };
```

Comment:
It is destructured here, but not in `onSavedQuerySaved`. Is it intentional?
I wouldn't use destructuring everywhere if it's not necessary )

Reply:
right you are! removed it is